verilog/formatting/formatter.cc

```diff
@@ -313,7 +313,24 @@ absl::Status FormatVerilogRange(const verible::TextStructureView& structure,
     return absl::CancelledError("Halting for diagnostic operation.");
   }
 
-  fmt.Emit(false, formatted_stream);
+  std::ostringstream output_buffer;
+  fmt.Emit(false, output_buffer);
+  *formatted_text = output_buffer.str();
+
+  // The range-format can output a spurious newline in the beginning (#1150).
+  // Whitespace handling needs some rework in the formatter, and it is not
+  // trivial in the current state to fix at the source.
+  // However, we can easily see the effects and mitigate it here: if we see a
+  // newline at the beginning of the formatted range, but no newline at the
+  // beginning of the original range, erase it in the formatted output.
+  // TODO(hzeller): This can go when whitespace handling is revisited.
+  //                (Emit(), FormatWhitespaceWithDisabledByteRanges())
+  const auto& text_lines = structure.Lines();
+  const char unformatted_begin = *text_lines[line_range.min - 1].begin();
+  if (!formatted_text->empty() && (*formatted_text)[0] == '\n' &&
+      unformatted_begin != '\n') {
+    formatted_text->erase(0, 1);
```

Comment:
Deleting from the front of a string could result in an expensive memcpy(), just beware.  (might be worth a comment)